File: java/src/main/java/com/penapereira/patterns/flyweight/ConcreteFlyweight.java
package com.penapereira.patterns.flyweight;

/** Stores intrinsic state; immutable, therefore sharable. */
public final class ConcreteFlyweight implements Flyweight {

    private final String intrinsicState;

    public ConcreteFlyweight(String intrinsicState) {
        this.intrinsicState = intrinsicState;
    }

    @Override
    public String operation(int extrinsicState) {
        return "ConcreteFlyweight(" + intrinsicState + ") with extrinsic state " + extrinsicState;
    }
}

File: python/src/patterns/flyweight/concrete_flyweight.py
class ConcreteFlyweight:
    """Stores intrinsic state; immutable, therefore sharable."""

    def __init__(self, intrinsic_state: str) -> None:
        self._intrinsic_state = intrinsic_state

    def operation(self, extrinsic_state: int) -> str:
        return f"ConcreteFlyweight({self._intrinsic_state}) with extrinsic state {extrinsic_state}"
